### DISCUSSION TITLE
Adding the same modules in the device twice, causing duplicate interfaces.

### DISCUSSION THREAD

**Discussion by milemar** [netbox-community/netbox#12258] [Q&A] [answered]
Hi,

Maybe someone can help me how to get around this.

I have a Synergy 12000 blade chassis with 6 module bays (1-6).
I have created a module type: Synergy 50Gb Interconnect Link module that has 2 interfaces: L1 and L2.
I have a chassis that has two those modules in bay 3 and 6.
Unfortunately Netbox doesn't allow me add a secondary module in bay 6 because the interfaces already exists (from the module in bay 3).
How to get around this? Do I need to create a duplicate module with slightly different interface name?

Thanks,
M

**Comment by milemar** [comment, marked as answer]
Sorry, found the solution using {module} in the interface name.

**Comment by mtinberg** [comment]
As far as this interconnect link, do the interfaces have names visible in the chassis management tools?  Are they configured in the unit like "InterconnectLink3/L1" and "InterconnectLink3/L2" because unless they are independent Devices there should be some way the device refers to them individually that probably includes the slot number.  You can template the slot number into interfaces names added from Modules using "{module}" in the name like https://demo.netbox.dev/dcim/module-types/1/interfaces/.  Even if that is not the case, you can create a unique naming convention for them using the module number templated into the interface name so they don't conflict and are individually identifiable.




—

Mark Tinberg ***@***.***>

Division of Information Technology-Network Services

University of Wisconsin-Madison


________________________________
From: milemar ***@***.***>
Sent: Friday, April 14, 2023 7:48 AM
To: netbox-community/netbox ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [netbox-community/netbox] Adding the same modules in the device causing duplicate interfaces. (Discussion #12258)


Hi,

Maybe someone can help me how to get around this.

I have a Synergy 12000 blade chassis with 6 module bays (1-6).
I have created a module type: Synergy 50Gb Interconnect Link module that has 2 interfaces: L1 and L2.
I have a chassis that has two those modules in bay 3 and 6.
Unfortunately Netbox doesn't allow me add a secondary module in bay 6 because the interfaces already exists (from the module in bay 3).
How to get around this? Do I need to create a duplicate module with slightly different interface name?

Thanks,
M
